### DISCUSSION TITLE
How do I speed up my api route?

### DISCUSSION THREAD

**Discussion by Kenramiscal1106** [vercel/next.js#36109] [Help]
I connected my **mongodb** database to my **api route in nextjs**. When I fetch it in the route _/mongo/documents_, it waits 2s (a little to long). I used **getServerSideProps** to fetch the data per request.

How can I speed up the waiting process of the route?

test site: [mongodb nextjs](https://next-js-one-psi.vercel.app/mongo/documents)
![image](https://user-images.githubusercontent.com/89060408/163072956-1c8651e7-8420-4171-a8b2-8e3ab7c6fbbc.png)

**Comment by GabenGar** [comment]
~~Obviously by rewriting your SQL query.~~ You have to set up redis cache and connection pooling although the latter won't make much difference if the page gets barely visited.
